hazelcast/src/main/java/com/hazelcast/config/AbstractCacheConfig.java

```diff
@@ -287,11 +297,57 @@ public HotRestartConfig getHotRestartConfig() {
 
     @Override
     public Class<K> getKeyType() {
-        return keyType;
+        return keyType != null ? keyType : resolveKeyType();
+    }
+
+    public String getKeyClassName() {
+        return keyClassName;
+    }
+
+    public CacheConfiguration<K, V> setKeyClassName(String keyClassName) {
+        this.keyClassName = keyClassName;
+        return this;
     }
 
     @Override
     public Class<V> getValueType() {
+        return valueType != null ? valueType : resolveValueType();
+    }
+
+    public String getValueClassName() {
+        return valueClassName;
+    }
+
+    public CacheConfiguration<K, V> setValueClassName(String valueClassName) {
+        this.valueClassName = valueClassName;
+        return this;
+    }
+
+    private Class<K> resolveKeyType() {
+        if (keyClassName != null) {
+            try {
+                keyType = (Class<K>) ClassLoaderUtil.loadClass(classLoader, keyClassName);
+            } catch (ClassNotFoundException e) {
+                throw new HazelcastException("Could not resolve key type", e);
+            }
+        }
+        if (keyType == null) {
+            keyType = (Class<K>) Object.class;
+        }
+        return keyType;
+    }
+
+    private Class<V> resolveValueType() {
```

Comment:
`resolveKeyType` and `resolveValueType` methods look similar with one another. Seems two method can be reduced to a single method to do same thing.